main_test.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"strings"
+	"testing"
+)
+
+type mockChangelogPrinter struct{}
+
+func (mcp *mockChangelogPrinter) content() (string, error) {
+	return "line 1\nline 2", nil
+}
+
+type mockTooManyChangelogPrinter struct{}
+
+func (mcp *mockTooManyChangelogPrinter) content() (string, error) {
+	return strings.Repeat("a", 25*1024), nil
+}
+
+func Test_processChangelog(t *testing.T) {
```

Comment:
the unit test reveals problems with the processChangelog function: no return value, can not directly assert